source/particle/world.cc

```diff
@@ -553,28 +558,64 @@ namespace aspect
                           ExcMessage("Periodic boundaries combined with particles currently "
                                      "only work with 90 degree opening angle in spherical shell."));
 
+              // Find out if we need to rotate the temporary properties
+              unsigned int property_index_to_rotate = numbers::invalid_unsigned_int;
+              if (Plugins::plugin_type_matches<Particle::Integrator::Euler<dim>>(*integrator) == true)
+                {
+                  // nothing to do
+                }
+              else if (Plugins::plugin_type_matches<Particle::Integrator::RK2<dim>>(*integrator) == true)
+                {
+                  // rotate original location
+                  property_index_to_rotate = get_property_manager().get_data_info().get_position_by_field_name("internal: integrator properties");
+                  AssertThrow(property_index_to_rotate != numbers::invalid_unsigned_int, ExcMessage("Could not find the property index."));
+                }
+              else
+                {
+                  AssertThrow(false,
+                              ExcMessage("Periodic boundaries in a spherical shell are not supported for this integration scheme."));
+                }
+
               typename ParticleHandler<dim>::particle_iterator particle = particle_handler->begin();
               for (; particle != particle_handler->end(); ++particle)
                 {
                   // modify the particle position if it crossed a periodic boundary
                   Point<dim> particle_position = particle->get_location();
 
+                  // define a rotation matrix for the new position depending on the boundary
+                  Tensor<2,dim> rotation_matrix;
+
                   if (particle_position[0] < 0.)
                     {
-                      const double temp = particle_position[0];
-                      particle_position[0] = particle_position[1];
-                      particle_position[1] = -temp;
+                      rotation_matrix[0][1] = 1.;
+                      rotation_matrix[1][0] = -1.;
                     }
                   else if (particle_position[1] < 0.)
                     {
-                      const double temp = particle_position[0];
-                      particle_position[0] = -particle_position[1];
-                      particle_position[1] = temp;
+                      rotation_matrix[0][1] = -1.;
+                      rotation_matrix[1][0] = 1.;
                     }
                   else
                     continue;
 
-                  particle->set_location(particle_position);
+                  const Tensor<1,dim> rotated_tensor (rotation_matrix * particle_position);
```

Comment:
call this `periodic_position` to make it clearer what this actually is:
```suggestion
                  const Tensor<1,dim> periodic_position = rotation_matrix * particle_position;
```